xml/deploy_cephadm.xml

```diff
@@ -701,131 +701,293 @@ o- time_server ................................................ [enabled]
+          Generate an SSL certificate:
+         </para>
+<screen>
+&prompt.root;openssl req -newkey rsa:4096 -nodes -sha256 \
+ -keyout /var/lib/registry/certs/domain.key -x509 -days 365 \
+ -out /var/lib/registry/certs/domain.crt
+</screen>
+         <note>
+          <para>
+           Set the <literal>CN=[value]</literal> value to the fully qualified
+           domain name of the host ([<option>REG_HOST_FQDN</option>]).
+          </para>
+         </note>
+        </step>
+        <step>
+         <para>
+          Copy the certificate to all cluster nodes and refresh the certificate
+          cache:
+         </para>
+<screen>
+&prompt.root;salt-cp '*' /var/lib/registry/certs/domain.crt \
+ /etc/pki/trust/anchors/
+&prompt.root;salt '*' cmd.shell "update-ca-certificates"
+</screen>
+        </step>
+        <step>
+         <para>
+          Generate a username and password combination for authentication to
+          the registry:
+         </para>
+<screen>
+&prompt.root;htpasswd2 -bBc /var/lib/registry/auth/htpasswd \
+ <option>REG_USERNAME</option> <option>REG_PASSWORD</option>
+</screen>
+        </step>
+        <step>
+         <para>
+          Start the secure registry:
+         </para>
+<screen>
+podman run --name myregistry -p <option>REG_HOST_PORT</option>:5000 \
+ -v /var/lib/registry:/var/lib/registry \
+ -v /var/lib/registry/auth:/auth:z \
+ -e "REGISTRY_AUTH=htpasswd" \
+ -e "REGISTRY_AUTH_HTPASSWD_REALM=Registry Realm" \
+ -e REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd \
+ -v /var/lib/registry/certs:/certs:z \
+ -e "REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt" \
+ -e "REGISTRY_HTTP_TLS_KEY=/certs/domain.key" \
+ -e REGISTRY_COMPATIBILITY_SCHEMA1_ENABLED=true -d registry:2
+</screen>
+        </step>
+        <step>
+         <para>
+          Test secure access to the registry:
+         </para>
+<screen>
+&prompt.user;curl https://<option>REG_HOST_FQDN</option>:<option>REG_HOST_PORT</option>/v2/_catalog \
+ -u <option>REG_USERNAME</option>:<option>REG_PASSWORD</option>
+</screen>
+        </step>
+       </substeps>
+      </step>
+     </procedure>
+     <procedure>
+      <title>Configure the local registry and access credentials</title>
+      <step>
+       <para>
+        Configure the URL of the local registry:
+       </para>
+<screen>&prompt.cephuser;ceph-salt config /containers/registry_auth/registry set <replaceable>REG_HOST_URL</replaceable></screen>
+      </step>
+      <step>
+       <para>
+        Configure the user name and password to access the local registry:
+       </para>
+<screen>&prompt.cephuser;ceph-salt config /containers/registry_auth/username set <replaceable>REG_USERNAME</replaceable></screen>
+<screen>&prompt.cephuser;ceph-salt config /containers/registry_auth/password set <replaceable>REG_PASSWORD</replaceable></screen>
+      </step>
+      <step>
+       <para>
+        Run <command>ceph-salt apply</command> to update the &salt; pillar on
+        all minions.
+       </para>
```

Comment:
As with a previous comment, you only want to run `ceph-salt apply` as the last step of deployment, because that will bootstrap the cluster.